cloud/docker_client.go

```diff
@@ -158,24 +158,28 @@ func (c *dockerClientImpl) EnsureImageDownloaded(ctx context.Context, h *host.Ho
 		// Image already exists
 		return imageName, nil
 	} else if strings.Contains(err.Error(), "No such image") {
-		if settings.Method == distro.DockerImageBuildTypeImport {
-			err = c.importImage(ctx, h, imageName, settings.URL)
+		if options.Method == distro.DockerImageBuildTypeImport {
+			err = c.importImage(ctx, h, imageName, options.Image)
 			grip.Info(message.Fields{
 				"operation":     "EnsureImageDownloaded",
 				"details":       "import image",
 				"duration_secs": time.Since(start).Seconds(),
 			})
 			return imageName, errors.Wrap(err, "error importing image")
-		} else if settings.Method == distro.DockerImageBuildTypePull {
-			err = c.pullImage(ctx, h, settings.URL, settings.RegistryUser, settings.RegistryPassword)
+		} else if options.Method == distro.DockerImageBuildTypePull {
+			image := options.Image
+			if options.RegistryName != "" {
```

Comment:
The port is specified because that's the port hosting the docker API on localhost. For real registries, the URL will map to some ports on the host machine already, so it's not needed